src/modules/hooks/base_hooks.js

```diff
@@ -0,0 +1,55 @@
+module.exports = class BaseHooks {
```

Comment:
Please, above each hook method - add docs that explains what are the arguments that will be passed to it.